android/sdl_android/src/main/java/com/smartdevicelink/streaming/StreamPacketizer.java

```diff
@@ -270,9 +290,18 @@ private void sendByteBufferData(ByteBuffer data) {
 		buffer.flip();
 
 		try {
-			mOutputQueue.put(buffer);
+			mOutputQueue.put(new ByteBufferWithListener(buffer, completionListener));
 		} catch (InterruptedException e) {
 			Thread.currentThread().interrupt();
 		}
 	}
+
+	private class ByteBufferWithListener{

```

Comment:
this class is in base, do we need it here as well?

Reply:
Maybe not. But it is public and cannot be removed from base without a breaking change.